components/ConvertStepper/Step.js

```diff
@@ -0,0 +1,192 @@
+`
+
+const pulseAnimation = css`
+  animation: ${keyframes`
+    from {
+      opacity: 1;
+    }
+
+    to {
+      opacity: 0.25;
+    }
+  `} 0.75s linear alternate infinite;
+`
+
+function getBorderColor(status) {
+  switch (status) {
+    case 'success':
+      return '#2CC68F'
+    case 'error':
+      return '#FF7163'
+    case 'working':
+      return '#FFAA75'
+    case 'waiting':
+      return '#FFAA75'
+    default:
+      return 'transparent'
+  }
+}
+
+function Step({ title, status, dormant, number, className, transactionHash }) {
+  const desc = useMemo(() => STATUS_DESC[status], [status])
+  const borderColor = useMemo(() => getBorderColor(status), [status])
+
+  return (
+    <div
+      className={className}
+      css={`
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+      `}
+    >
+      <div
+        css={`
+          width: 110px;
+          margin-bottom: 25px;
+        `}
+      >
+        <div
+          css={`
+            position: relative;
+            width: 100%;
+            padding-top: 100%;
+          `}
+        >
+          <div
+            css={`
+              ${ABSOLUTE_FILL}
+
+              display: flex;
+
+              align-items: center;
+              justify-content: center;
+            `}
+          >
+            {status === 'waiting' && (
+              <span
+                css={`
+                  position: absolute;
+
+                  top: 50%;
+                  left: 50%;
+
+                  transform: translate(-50%, -50%);
+
+                  line-height: 1;
+                  color: #ffffff;
+                  font-size: 24px;
+                  font-weight: 600;
+
+                  z-index: 1;
+                `}
+              >
+                {number}
+              </span>
+            )}
+
+            <StatusIcon status={status} />
+
+            <div
+              css={`
+                ${ABSOLUTE_FILL}
+
+                border-radius: 100%;
+
+                border: 2px solid ${dormant ? 'transparent' : borderColor};
+
+                ${status === 'waiting' && pulseAnimation}
+                ${status === 'working' && spinAnimation}
+              `}
+            ></div>
+          </div>
+        </div>
+      </div>
+      <h2
+        css={`
+          text-align: center;
+          margin-bottom: 7px;
+          font-size: 20px;
+          color: ${status === 'error' ? '#FF7C7C' : '#4A5165'};
+          font-weight: 500;
+        `}
+      >
+        {title}
+      </h2>
+
+      <p
+        css={`
+          text-align: center;
+          margin-bottom: 0;
+          font-size: 14px;
+          color: ${status === 'success' ? '#2CC68F' : '#637381'};
+        `}
+      >
+        {desc}
+      </p>
+
+      <div
+        css={`
+          position: relative;
+
+          /* Avoid visual jump when showing tx by pre-filling space */
+          padding-bottom: 40px;
+          margin-bottom: -40px;
```

Comment:
I think this is totally fine, but just wondering if it would be better to use `visibility: none` to pre-fill the space, in case `TransactionBadge` ever changes size?

I remember doing an accessibility test on macOS before with the screen reader, to make sure it wouldn't read items in `visibility: none`, but we might have to double check.

Reply:
That can work providing some type of character is provided to match the height exactly, In the past I've used &nbsp; but I heard a rumble somewhere that this can be announced by certain readers sometimes so I've steered away from it.

Reply:
I was thinking about an empty transaction hash, e.g. `'0x'`, but agree it's also not that nice. I wonder if using a `min-height` would allow us to do this in one styling declaration, but otherwise, let's keep it as is!